admin/src/js/controllers/upgrade.js

```diff
@@ -164,13 +178,30 @@ angular.module('controllers').controller('UpgradeCtrl',
       }).catch(() => {});
     };
 
+    const waitUntilApiStarts = () => {
+      return $http
+        .get(POLL_URL)
+        .catch(() => {
+          const waitOneSecond = () => new Promise(r => $timeout(r, 1000));
+          return waitOneSecond().then(() => waitUntilApiStarts());
```

Comment:
Will this make an increasingly long promise chain? Can we use a callback, (ie: pass in a Promise.resolve function) so we don't keep chaining?